File: practice/mathematics/fundamentals/sherlock_and_moving_tiles/sherlock_and_moving_tiles.hs
module Main where

import Control.Monad (forM_)
import Text.Printf

main :: IO ()
main = do
    c <- getLine
    let [l, s1, s2] = map (read :: String -> Double) $ words c
        s = sqrt(2.0) * l
        t = abs $ s1 - s2
    q <- readLn :: IO Int
    forM_ [1..q] $ \_ -> do
        qi <- readLn :: IO Double
        let r = sqrt(2.0 * qi)
            ans = (abs (r - s)) / t
        printf "%.6f\n" ans

File: practice/mathematics/fundamentals/sherlock_and_moving_tiles/sherlock_and_moving_tiles.cpp
#include <cmath>
#include <iostream>
#include <iomanip> 

using namespace std;

int main()
{
    long long q, qi;
    double L, s1, s2;
    
    cin >> L >> s1 >> s2;
    double s = sqrt(2.0) * L;
    double t = s1 - s2;
    
    cin >> q;
    
    while (q--)
    {
        cin >> qi;
        double r = sqrt(2.0 * qi);
        double ans = abs(r - s) / abs(t);
        cout << setprecision(6) << fixed << ans << endl;
    }
        
    return 0;
}

